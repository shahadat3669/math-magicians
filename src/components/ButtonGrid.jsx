import { Component } from 'react';
import PropTypes from 'prop-types';

class ButtonGrid extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    const { clickHandler } = this.props;
    return (
      <div className="key-grid">
        <button type="button" className="button" onClick={clickHandler}>
          AC
        </button>
        <button type="button" className="button" onClick={clickHandler}>
          +/-
        </button>
        <button type="button" className="button" onClick={clickHandler}>
          %
        </button>
        <button
          type="button"
          className="button button--orange"
          onClick={clickHandler}
        >
          รท
        </button>
        <button type="button" className="button" onClick={clickHandler}>
          7
        </button>
        <button type="button" className="button" onClick={clickHandler}>
          8
        </button>
        <button type="button" className="button" onClick={clickHandler}>
          9
        </button>
        <button
          type="button"
          className="button button--orange"
          onClick={clickHandler}
        >
          x
        </button>
        <button type="button" className="button" onClick={clickHandler}>
          4
        </button>
        <button type="button" className="button" onClick={clickHandler}>
          5
        </button>
        <button type="button" className="button" onClick={clickHandler}>
          6
        </button>
        <button
          type="button"
          className="button button--orange"
          onClick={clickHandler}
        >
          -
        </button>
        <button type="button" className="button" onClick={clickHandler}>
          1
        </button>
        <button type="button" className="button" onClick={clickHandler}>
          2
        </button>
        <button type="button" className="button" onClick={clickHandler}>
          3
        </button>
        <button
          type="button"
          className="button button--orange"
          onClick={clickHandler}
        >
          +
        </button>
        <button
          type="button"
          className="button button--zero"
          onClick={clickHandler}
        >
          0
        </button>
        <button type="button" className="button" onClick={clickHandler}>
          .
        </button>
        <button
          type="button"
          className="button button--orange"
          onClick={clickHandler}
        >
          =
        </button>
      </div>
    );
  }
}
export default ButtonGrid;

ButtonGrid.propTypes = {
  clickHandler: PropTypes.func.isRequired,
};
