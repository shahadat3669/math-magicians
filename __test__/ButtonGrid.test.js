import React from 'react';
import renderer from 'react-test-renderer';

import ButtonGrid from '../src/components/ButtonGrid';

it('renders correctly when there are no items', () => {
  const tree = renderer.create(<ButtonGrid />).toJSON();
  expect(tree).toMatchSnapshot();
});
