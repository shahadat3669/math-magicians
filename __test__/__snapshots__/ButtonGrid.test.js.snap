// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are no items 1`] = `
<div
  className="key-grid"
>
  <button
    className="button"
    onClick={[Function]}
    type="button"
  >
    AC
  </button>
  <button
    className="button"
    onClick={[Function]}
    type="button"
  >
    +/-
  </button>
  <button
    className="button"
    onClick={[Function]}
    type="button"
  >
    %
  </button>
  <button
    className="button button--orange"
    onClick={[Function]}
    type="button"
  >
    รท
  </button>
  <button
    className="button"
    onClick={[Function]}
    type="button"
  >
    7
  </button>
  <button
    className="button"
    onClick={[Function]}
    type="button"
  >
    8
  </button>
  <button
    className="button"
    onClick={[Function]}
    type="button"
  >
    9
  </button>
  <button
    className="button button--orange"
    onClick={[Function]}
    type="button"
  >
    x
  </button>
  <button
    className="button"
    onClick={[Function]}
    type="button"
  >
    4
  </button>
  <button
    className="button"
    onClick={[Function]}
    type="button"
  >
    5
  </button>
  <button
    className="button"
    onClick={[Function]}
    type="button"
  >
    6
  </button>
  <button
    className="button button--orange"
    onClick={[Function]}
    type="button"
  >
    -
  </button>
  <button
    className="button"
    onClick={[Function]}
    type="button"
  >
    1
  </button>
  <button
    className="button"
    onClick={[Function]}
    type="button"
  >
    2
  </button>
  <button
    className="button"
    onClick={[Function]}
    type="button"
  >
    3
  </button>
  <button
    className="button button--orange"
    onClick={[Function]}
    type="button"
  >
    +
  </button>
  <button
    className="button button--zero"
    onClick={[Function]}
    type="button"
  >
    0
  </button>
  <button
    className="button"
    onClick={[Function]}
    type="button"
  >
    .
  </button>
  <button
    className="button button--orange"
    onClick={[Function]}
    type="button"
  >
    =
  </button>
</div>
`;
